app/forms/state_file/verification_code_form.rb

```diff
@@ -0,0 +1,33 @@
+module StateFile
+  class VerificationCodeForm < QuestionsForm
+    set_attributes_for :misc, :verification_code
+
+    def valid?
+      # Delegate validation to a phone or email verification form.
+      # If there's no contact preference, use the email form.
+      validation_form_class = is_text_based? ? PhoneVerificationForm : EmailVerificationForm
+      form = validation_form_class.new(intake, attributes_for(:misc))
+      is_valid = form.valid?
+      unless is_valid
+        errors.add(:verification_code, form.errors[:verification_code])
+      end
+
+      is_valid
+    end
+
+    def save
+      intake.touch(:phone_number_verified_at) if is_text_based?
+      intake.touch(:email_address_verified_at) if is_email_based?
+    end
+
+    private
+
+    def is_text_based?
```

Comment:
I like that you separated this out! Makes it easier to read. I would prefer using the word `sms` over `text` however since technically both email and texting texts are "text based"